### DISCUSSION TITLE
Rendering Animation with object matrix output transforms continuously

### DISCUSSION THREAD

**Discussion by Chryfi** [JacquesLucke/animation_nodes#1848] [Q&A] [answered]
I bake the action to an empty which is at the root position of my sword. I want to create a trail for which I delay the actions of that empty onto several other empties which function as hooks for the trail plane. This is my node setup:
![Screenshot_327](https://user-images.githubusercontent.com/71967555/163427008-6be520b7-d3fc-4bad-811d-cfb29a3d2767.png)

To reduce manual effort I translate them using an offset matrix node (with translation mode set to local). This is needed to translate the empties to the tip and the middle of the sword. Everything seems to work, but when I enable motion blur and render (both cycles and Eevee have this problem) the empties which I transformed fly away as can be seen here. This also happens without motionblur when rendering the animation, which makes this unusable.
![Screenshot_328](https://user-images.githubusercontent.com/71967555/163427516-8476fa8d-ff7a-4785-8a9e-cb5c59a55274.png)
This only happens during rendering, after it is finished they will reset back to the position they were supposed to be. It also happens when rendering vector pass.

EDIT: Another thing I discovered: when rendering in viewport and clicking somewhere in the viewport, they also everytime I click fly away one step.

**Comment by OmarEmaraDev** [comment]
Can you share a minimal file that reproduces the issue?

**Comment by Chryfi** [comment]
[mesh blur bug.zip](https://github.com/JacquesLucke/animation_nodes/files/8491946/mesh.blur.bug.zip)
I used Blender 3.1 with the latest build release of this plugin.

**Comment by OmarEmaraDev** [comment, marked as answer]
The issue is that you are setting the location of objects, reading their locations right after, then setting their location again. This is unreliable because Blender only updates the scene after AN finishes executing entirely. So you should generally set the location of objects only once and not read them again in the node tree. ([There is a workaround, but using it is highly discouraged!](https://docs.animation-nodes.com/documentation/nodes/object/evaluate_object/))

The best way to do what you want is to use the newly added Evaluate FCurves Transforms node as follows:

![20220414-233258](https://user-images.githubusercontent.com/25300994/163480962-3fe203e8-1b27-497b-a729-3d17d355d40f.png)

**Comment by Chryfi** [comment]
Am I doing something wrong?
![Screenshot_329](https://user-images.githubusercontent.com/71967555/163573830-c64e502b-7f3a-4d69-8260-2d88bfbee30e.png)

**Reply by OmarEmaraDev** [reply]
The evaluate node is not connected to the input parameter of the loop?

**Reply by Chryfi** [reply]
I do not understand... I went to Loop input and clicked on list generator to create a matrix list and I connected it with the evaluate FCurves Transforms node. Isn't this what you also did in the screenshot?

**Reply by OmarEmaraDev** [reply]
@Chryfi I mean connect the FCurves list parameter of the loop input node to the FCurves evaluate node.

**Reply by Chryfi** [reply]
Sorry for the late replay. I made it work now, you were right, however.... now the delay is.... not delayed but in the future? I dont know how to properly describe it, see for yourself
![Screenshot_342](https://user-images.githubusercontent.com/71967555/165294807-9e56a942-416f-439a-b7ae-e4445c7f3b8b.png)
![Screenshot_341](https://user-images.githubusercontent.com/71967555/165294811-54b76220-738f-4f5d-8fb5-e7312f7a63fc.png)

**Reply by Chryfi** [reply]
Okay I think I got the hang of it, is this correct? I multiply the offset by -1 to get a delay?
![Screenshot_343](https://user-images.githubusercontent.com/71967555/165310588-2af617b4-6f7f-493b-8b9a-8bc6325acf9c.png)

**Reply by OmarEmaraDev** [reply]
@Chryfi Yes, multiplying by a negative factor is the right choice here.
